types/generic.d.ts

```diff
@@ -1,6 +1,16 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License.
 
+import { FunctionOptions, RetryOptions } from './index';
+
+export interface GenericFunctionOptions extends FunctionOptions {
+    /**
+     * An optional retry policy to rerun a failed execution until either successful completion occurs or the maximum number of retries is reached.
+     * Learn more [here](https://learn.microsoft.com/azure/azure-functions/functions-bindings-error-pages)
+     */
+    retry?: RetryOptions;
+}
+
```

Comment:
Is there a reason this was not just added to all function options?

Reply:
It's not supported in all trigger types. You can see the list of supported types here: https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-error-pages#retries. Specifically this stuff only applies when the "Retry Source" is "Retry policies" in the table shown at that link